### PR TITLE
Add test placeholders for UDP/TCP Mencius.

### PR DESCRIPTION
**Pull request by TheJMaster**

### CODE REVIEW

File: benchmark/Benchmarks.cpp [hunk 1]
```diff
@@ -1,18 +1,97 @@
...
 
-static void BM_StringCreation(benchmark::State& state) {
+/* Benchmark Fixtures */
+
+class TCPMenciusFixture : public benchmark::Fixture {
+public:
+  void SetUp(const ::benchmark::State& state) {
+  }
+
+  void TearDown(const ::benchmark::State& state) {
```
Comment (LukasJoswiak) [inline benchmark/Benchmarks.cpp]: &state instead of & state (fix elsewhere in file as well).

File: benchmark/Benchmarks.cpp [hunk 2]
```diff
@@ -1,18 +1,97 @@
 #include <benchmark/benchmark.h>
 
-static void BM_StringCreation(benchmark::State& state) {
+/* Benchmark Fixtures */
+
+class TCPMenciusFixture : public benchmark::Fixture {
+public:
```
Comment (LukasJoswiak) [inline benchmark/Benchmarks.cpp]: Indent by one space.

**Review by LukasJoswiak — CHANGES_REQUESTED** [review]

File: benchmark/Benchmarks.cpp [hunk 3]
```diff
@@ -1,18 +1,97 @@
...
+  }
+
+  void TearDown(const ::benchmark::State& state) {
+  }
+};
+
+/* Benchmark Arguments */
+
+static void TestArguments(benchmark::internal::Benchmark* b) {
+  for (int i = 3; i <= 5; i += 2) {
```
Comment (LukasJoswiak) [inline benchmark/Benchmarks.cpp]: Magic numbers.

**Review by LukasJoswiak — APPROVED** [review]

File: benchmark/Benchmarks.cpp [hunk 4]
```diff
@@ -1,18 +1,103 @@
 #include <benchmark/benchmark.h>
 
-static void BM_StringCreation(benchmark::State& state) {
+/* Constants */
+const int NUM_REPLICAS[] = {3, 5};
```
Comment (LukasJoswiak) [inline benchmark/Benchmarks.cpp]: Wrong naming conventions

**Review by LukasJoswiak — CHANGES_REQUESTED** [review]

File: benchmark/Benchmarks.cpp [hunk 5]
```diff
@@ -1,18 +1,103 @@
...
+}
+
+/* Benchmarks */
+
+BENCHMARK_DEFINE_F(TCPMenciusFixture, NoFailures)(benchmark::State &state) {
+  auto num_replicas = state.range(0);
+  auto num_clients = state.range(1);
+  auto drop_rate = state.range(2) / DROP_RATE_DENOM;
+
+  for (auto _ : state)
```
Comment (LukasJoswiak) [inline benchmark/Benchmarks.cpp:48]: Use curly braces.

File: benchmark/Benchmarks.cpp [hunk 6]
```diff
@@ -1,18 +1,103 @@
...
+    std::string empty_string;
+}
+BENCHMARK_REGISTER_F(TCPMenciusFixture, NoFailures)->Apply(TestArguments);
+
+BENCHMARK_DEFINE_F(TCPMenciusFixture, GradualFailures)(benchmark::State &state) {
+  auto num_replicas = state.range(0);
+  auto num_clients = state.range(1);
+  auto drop_rate = state.range(2) / DROP_RATE_DENOM;
+
   for (auto _ : state)
```
Comment (LukasJoswiak) [inline benchmark/Benchmarks.cpp:58]: See above.

File: benchmark/Benchmarks.cpp [hunk 7]
```diff
@@ -1,18 +1,103 @@
...
+
+BENCHMARK_DEFINE_F(TCPMenciusFixture, InstantFailures)(benchmark::State &state) {
+  auto num_replicas = state.range(0);
+  auto num_clients = state.range(1);
+  auto drop_rate = state.range(2) / DROP_RATE_DENOM;
 
-// Define another benchmark
-static void BM_StringCopy(benchmark::State& state) {
-  std::string x = "hello";
   for (auto _ : state)
```
Comment (LukasJoswiak) [inline benchmark/Benchmarks.cpp:68]: See above.

File: benchmark/Benchmarks.cpp (hunk 5 above)
Comment (TheJMaster) [reply benchmark/Benchmarks.cpp:48]: this will be deleted.

File: benchmark/Benchmarks.cpp [hunk 8]
```diff
@@ -1,18 +1,103 @@
...
+    std::string empty_string;
+}
+BENCHMARK_REGISTER_F(TCPMenciusFixture, InstantFailures)->Apply(TestArguments);
+
+BENCHMARK_DEFINE_F(UDPMenciusFixture, NoFailures)(benchmark::State &state) {
+  auto num_replicas = state.range(0);
+  auto num_clients = state.range(1);
+  auto drop_rate = state.range(2) / DROP_RATE_DENOM;
+
+  for (auto _ : state)
```
Comment (LukasJoswiak) [inline benchmark/Benchmarks.cpp:78]: See above.

File: benchmark/Benchmarks.cpp (hunk 7 above)
Comment (TheJMaster) [reply benchmark/Benchmarks.cpp:68]: this will be deleted

File: benchmark/Benchmarks.cpp (hunk 6 above)
Comment (TheJMaster) [reply benchmark/Benchmarks.cpp:58]: this will be deleted.

File: benchmark/Benchmarks.cpp (hunk 8 above)
Comment (TheJMaster) [reply benchmark/Benchmarks.cpp:78]: this will be deleted.

File: benchmark/Benchmarks.cpp [hunk 9]
```diff
@@ -1,18 +1,103 @@
...
-static void BM_StringCopy(benchmark::State& state) {
-  std::string x = "hello";
   for (auto _ : state)
-    std::string copy(x);
+    std::string empty_string;
+}
+BENCHMARK_REGISTER_F(TCPMenciusFixture, InstantFailures)->Apply(TestArguments);
+
+BENCHMARK_DEFINE_F(UDPMenciusFixture, NoFailures)(benchmark::State &state) {
+  auto num_replicas = state.range(0);
```
Comment (LukasJoswiak) [inline benchmark/Benchmarks.cpp:74]: Can you move this duplicate code to a helper function?

File: benchmark/Benchmarks.cpp [hunk 10]
```diff
@@ -1,18 +1,103 @@
...
+    std::string empty_string;
+}
+BENCHMARK_REGISTER_F(UDPMenciusFixture, NoFailures)->Apply(TestArguments);
+
+BENCHMARK_DEFINE_F(UDPMenciusFixture, GradualFailures)(benchmark::State &state) {
+  auto num_replicas = state.range(0);
+  auto num_clients = state.range(1);
+  auto drop_rate = state.range(2) / DROP_RATE_DENOM;
+
+  for (auto _ : state)
```
Comment (LukasJoswiak) [inline benchmark/Benchmarks.cpp:88]: See above.

File: benchmark/Benchmarks.cpp [hunk 11]
```diff
@@ -1,18 +1,103 @@
...
+    std::string empty_string;
+}
+BENCHMARK_REGISTER_F(UDPMenciusFixture, GradualFailures)->Apply(TestArguments);
+
+BENCHMARK_DEFINE_F(UDPMenciusFixture, InstantFailures)(benchmark::State &state) {
+  auto num_replicas = state.range(0);
+  auto num_clients = state.range(1);
+  auto drop_rate = state.range(2) / DROP_RATE_DENOM;
+
+  for (auto _ : state)
```
Comment (LukasJoswiak) [inline benchmark/Benchmarks.cpp:98]: See above.

File: benchmark/Benchmarks.cpp (hunk 10 above)
Comment (TheJMaster) [reply benchmark/Benchmarks.cpp:88]: this will be deleted.

File: benchmark/Benchmarks.cpp (hunk 11 above)
Comment (TheJMaster) [reply benchmark/Benchmarks.cpp:98]: this will be deleted.

**Review by LukasJoswiak — APPROVED** [review]

**Pull request merged**